File: WorldFeed/Server/src/WorldFeed.Common/Models/TwitterEntities/RelationshipState.cs
namespace WorldFeed.Common.Models.TwitterEntities
{

    using global::WorldFeed.Common.Public.Models.Interfaces;
    using global::WorldFeed.Common.Public.Models.Interfaces.DTO;

    public class RelationshipState : IRelationshipState
    {

        public RelationshipState(IRelationshipStateDTO relationshipStateDTO)
        {
            RelationshipStateDTO = relationshipStateDTO;
        }

        public IRelationshipStateDTO RelationshipStateDTO { get; set; }

        public long TargetId => RelationshipStateDTO.TargetUserId;

        public string TargetIdStr => RelationshipStateDTO.TargetUserIdStr;

        public string TargetName => RelationshipStateDTO.TargetUserName;

        public string TargetScreenName => RelationshipStateDTO.TargetUserScreenName;

        public bool Following => RelationshipStateDTO.Following;

        public bool FollowedBy => RelationshipStateDTO.FollowedBy;

        public bool FollowingRequested => RelationshipStateDTO.FollowingRequested;

        public bool FollowingRequestReceived => RelationshipStateDTO.FollowingRequestReceived;
    }
}

File: Sharebook/Client/src/app/core/Core/Models/TwitterEntities/RelationshipState.ts
import {IRelationshipStateDTO} from "../../../Public/Models/Interfaces/DTO/IRelationshipStateDTO";
import {IRelationshipState} from "../../../Public/Models/Interfaces/IRelationshipState";

export class RelationshipState implements IRelationshipState {
  constructor(relationshipStateDTO: IRelationshipStateDTO) {
    this.relationshipStateDTO = relationshipStateDTO;
  }

  public relationshipStateDTO: IRelationshipStateDTO;

  get targetId(): number {
    return this.relationshipStateDTO.targetUserId;
  }

  get targetIdStr(): string {
    return this.relationshipStateDTO.targetUserIdStr;
  }

  get targetName(): string {
    return this.relationshipStateDTO.targetUserName;
  }

  get targetScreenName(): string {
    return this.relationshipStateDTO.targetUserScreenName;
  }

  get following(): boolean {
    return this.relationshipStateDTO.following;
  }

  get followedBy(): boolean {
    return this.relationshipStateDTO.followedBy;
  }

  get followingRequested(): boolean {
    return this.relationshipStateDTO.followingRequested;
  }

  get followingRequestReceived(): boolean {
    return this.relationshipStateDTO.followingRequestReceived;
  }
}
